druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -544,12 +557,15 @@ protected QuerySpec getQuery(RelDataType rowType, RexNode filter, List<RexNode>
         for (int groupKey : groupSet) {
           final String s = fieldNames.get(groupKey);
           if (s.equals(druidTable.timestampFieldName)) {
-            granularity = Granularity.NONE;
-            builder.add(s);
+            granularity = Granularity.ALL;
+            timeExtractionDimensionSpec = TimeExtractionDimensionSpec.makeFullTimeExtract();
+            builder.add(DruidConnectionImpl.DEFAULT_RESPONSE_TIMESTAMP_COLUMN);
+            dimensions.add(timeExtractionDimensionSpec);
+            //@TODO remove this assert cause we can have multiple extract on time column
```

Comment:
This is a good catch, could you create a follow-up? I guess if you group by two different extract functions right now... we fail with an assertion error... right? That is bad, at least we should detect this in DruidRules so we do not hit the assert.